flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/exceptionhistory/ExceptionHistoryEntryExtractorTest.java

```diff
@@ -0,0 +1,159 @@
+                IsIterableContainingInOrder.contains(
+                        ExceptionHistoryEntryMatcher.matchesFailure(
+                                otherException,
+                                otherFailureTimeframe.lowerEndpoint(),
+                                otherFailureTimeframe.upperEndpoint(),
+                                otherExecutionVertex.getTaskNameWithSubtaskIndex(),
+                                otherExecutionVertex.getCurrentAssignedResourceLocation())));
+    }
+
+    @Test
+    public void extractGlobalFailure() {
+        final Throwable otherException0 = new RuntimeException("Expected other failure #0");
+        final ExecutionVertex otherExecutionVertex0 = extractExecutionVertex(0);
+        final Range<Long> otherFailureTimeframe0 =
+                triggerFailure(otherExecutionVertex0, otherException0);
+
+        final Throwable otherException1 = new IllegalStateException("Expected other failure");
+        final ExecutionVertex otherExecutionVertex1 = extractExecutionVertex(1);
+        final Range<Long> otherFailureTimeframe1 =
+                triggerFailure(otherExecutionVertex1, otherException1);
+
+        final Throwable rootCause = new Exception("Expected root failure");
+        final long rootTimestamp = System.currentTimeMillis();
+        final ExceptionHistoryEntryExtractor testInstance = new ExceptionHistoryEntryExtractor();
+        final RootExceptionHistoryEntry actualEntry =
+                testInstance.extractGlobalFailure(
+                        executionGraph.getAllExecutionVertices(), rootCause, rootTimestamp);
+
+        assertThat(
+                actualEntry,
+                ExceptionHistoryEntryMatcher.matchesGlobalFailure(rootCause, rootTimestamp));
+        assertThat(
+                actualEntry.getConcurrentExceptions(),
+                IsIterableContainingInOrder.contains(
+                        ExceptionHistoryEntryMatcher.matchesFailure(
+                                otherException0,
+                                otherFailureTimeframe0.lowerEndpoint(),
+                                otherFailureTimeframe0.upperEndpoint(),
+                                otherExecutionVertex0.getTaskNameWithSubtaskIndex(),
+                                otherExecutionVertex0.getCurrentAssignedResourceLocation()),
+                        ExceptionHistoryEntryMatcher.matchesFailure(
+                                otherException1,
+                                otherFailureTimeframe1.lowerEndpoint(),
+                                otherFailureTimeframe1.upperEndpoint(),
+                                otherExecutionVertex1.getTaskNameWithSubtaskIndex(),
+                                otherExecutionVertex1.getCurrentAssignedResourceLocation())));
+    }
+
+    private Range<Long> triggerFailure(ExecutionVertex executionVertex, Throwable throwable) {
+        final long startInMillis = System.currentTimeMillis();
```

Comment:
shouldn't be necessary; the failure timestamp is the FAILED state timestamp of the vertex

Reply:
Good point 👍